File: nodejs/base/api/lib/routing/defaults/default-route-handler-configurator.ts
import { McmaApiRouteHandler } from "../route-handler";
import { RouteStartedHandler } from "./route-started-handler";
import { RouteCompletedHandler } from "./route-completed-handler";
import { DefaultRouteHandlerBuilder } from "./default-route-handler-builder";

export class DefaultRouteHandlerConfigurator<T> {
    onStarted: RouteStartedHandler;
    onCompleted: RouteCompletedHandler<T>;
    
    constructor(private handlerBuilder: DefaultRouteHandlerBuilder<T>) { }

    create(): McmaApiRouteHandler {
        return this.handlerBuilder(this.onStarted, this.onCompleted);
    }
}

File: dotnet/base/Mcma.Api/Routing/Defaults/DefaultRouteHandlerConfigurator.cs
using System.Threading.Tasks;
using System;

namespace Mcma.Api.Routing.Defaults
{
    internal class DefaultRouteHandlerConfigurator<TResult>
    {
        internal DefaultRouteHandlerConfigurator(DefaultRouteHandlerBuilder<TResult> handlerBuilder)
        {
            HandlerBuilder = handlerBuilder;
        }

        private DefaultRouteHandlerBuilder<TResult> HandlerBuilder { get; }

        internal Func<McmaApiRequestContext, Task> OnStarted { get; set; }

        internal Func<McmaApiRequestContext, TResult, Task> OnCompleted { get; set; }

        internal Func<McmaApiRequestContext, Task> Create() => HandlerBuilder(OnStarted, OnCompleted);
    }
}
